R/estimate_delay.R

```diff
@@ -48,13 +48,29 @@ estimate_delay <- function(
     reporting_triangle,
     max_delay = ncol(reporting_triangle) - 1,
     n = nrow(reporting_triangle)) {
+  if (max_delay > ncol(reporting_triangle) - 1) {
+    message(sprintf(
+      "The maximum delay must be less than the number of columns in the reporting triangle. The maximum delay will be set to %d.", # nolint
+      ncol(reporting_triangle) - 1
+    ))
+    max_delay <- ncol(reporting_triangle) - 1
+  }
+
+  if (max_delay < ncol(reporting_triangle) - 1 && max_delay > 0) {
```

Comment:
This will be quite an annoying message in practice. Is it needed or if it is is it needed for every single time it comes up i.e. do you want it once persession

Reply:
Probably once per session -- can change

Reply:
changed